next-app/src/pages/index.tsx

```diff
@@ -0,0 +1,122 @@
+import Head from 'next/head';
+import Image from 'next/image';
+import { Inter } from '@next/font/google';
+import styles from '@/styles/Home.module.css';
+import dynamic from 'next/dynamic';
+import Component2 from 'lib2/component';
+
+const inter = Inter({ subsets: ['latin'] });
+const Component = dynamic(() => import('lib2/component'), {
+    ssr: false,
+});
+
+export default function Home() {
+    return (
+        <>
+            <Head>
+                <title>Create Next App</title>
+                <meta name="description" content="Generated by create next app" />
+                <meta name="viewport" content="width=device-width, initial-scale=1" />
+                <link rel="icon" href="/favicon.ico" />
+            </Head>
+            <main className={styles.main}>
+                <div className={styles.description}>
+                    <p>
+                        Get started by editing&nbsp;
+                        <code className={styles.code}>pages/index.tsx</code>
+                    </p>
+                    <div>
+                        <a
+                            href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
+                            target="_blank"
+                            rel="noopener noreferrer"
+                        >
+                            By{' '}
+                            <Image
+                                src="/vercel.svg"
+                                alt="Vercel Logo"
+                                className={styles.vercelLogo}
+                                width={100}
+                                height={24}
+                                priority
+                            />
+                        </a>
+                    </div>
+                </div>
+                <Component>Hello client rendered</Component>
+                <Component2>Hello server rendered</Component2>
```

Comment:
Rendering both the server rendered and client-only-rendered verisons of the component. If oyu downgrade styled-components, only the client-only-rendered one will work, the other one will break the app.